cmd/genesis/command.go

```diff
@@ -0,0 +1,304 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/kowala-tech/kcoin/accounts/abi"
+	"github.com/kowala-tech/kcoin/common"
+	"github.com/kowala-tech/kcoin/contracts/network/contracts"
+	"github.com/kowala-tech/kcoin/core"
+	"github.com/kowala-tech/kcoin/core/vm"
+	"github.com/kowala-tech/kcoin/core/vm/runtime"
+	"github.com/kowala-tech/kcoin/params"
+	"github.com/pkg/errors"
+	"io"
+	"math/big"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+var (
+	ErrEmptyMaxNumValidators                        = errors.New("max number of validators is mandatory")
+	ErrEmptyUnbondingPeriod                         = errors.New("unbonding period in days is mandatory")
+	ErrEmptyWalletAddressValidator                  = errors.New("Wallet address of genesis validator is mandatory")
+	ErrInvalidWalletAddressValidator                = errors.New("Wallet address of genesis validator is invalid")
+	ErrEmptyPrefundedAccounts                       = errors.New("empty prefunded accounts, at least the validator wallet address should be included")
+	ErrWalletAddressValidatorNotInPrefundedAccounts = errors.New("prefunded accounts should include genesis validator account")
+	ErrInvalidAddressInPrefundedAccounts            = errors.New("address in prefunded accounts is invalid")
+	ErrInvalidContractsOwnerAddress                 = errors.New("address used for smart contracts is invalid")
+)
+
+type GenerateGenesisCommand struct {
```

Comment:
Why make this type exportable with only non-exported members? Maybe I'm misunderstanding the use cases for it, but shouldn't it be either an interface or have at least one exported member?